ui/core/proto_utils/gear.ts

```diff
@@ -163,6 +163,13 @@ export class Gear extends BaseGear {
 			.includes(itemId);
 	}
 
+	hasTrinketFromOptions(itemIds: number[]): boolean {
+		return this.getTrinkets()
+			.filter(t => !!t)
```

Comment:
A nicer way of doing this would be:
```
.filter((t): t is EquippedItem => !!t)
```
So you don't have to say `t!.item.id` but you just handle the type guard better